expfmt/text_parse.go

```diff
@@ -125,6 +126,15 @@ func (p *TextParser) reset(in io.Reader) {
 	} else {
 		p.buf.Reset(in)
 	}
+
+	// Some clients like to throw in UTF-8 BOM chars
+	maybeBOM, err := p.buf.Peek(3)
+	if len(maybeBOM) == 3 && maybeBOM[0] == 0xEF && maybeBOM[1] == 0xBB && maybeBOM[2] == 0xBF && err == nil {
+		// The common case dealing with funny Windows clients
+		log.Debug("Discarding BOM")
```

Comment:
We don't use any logging anywhere in this package. It might come as a surprise to the (possibly very diverse) users, and it's pulling in another dependency. I think it's fine to silently swallow the BOMs.